### ISSUE TITLE
[Bug] AUTO PARTITION change

### ISSUE THREAD

**Issue by WuliHuazai** [apache/doris#32596]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

apache-doris-2.1.0-bin-x64

### What's Wrong?

Create Sql:
CREATE TABLE IF NOT EXISTS test_db.testMonth
(
	profileId bigint NOT NULL,
	reportDateTime date NOT NULL,
	keywordId bigint NOT NULL,
        ....
)
UNIQUE KEY(profileId,reportDateTime,keywordId)
AUTO PARTITION BY RANGE date_trunc(reportDateTime,'MONTH')
(
 ) 
DISTRIBUTED BY HASH(reportDateTime) BUCKETS 10
PROPERTIES (
  "dynamic_partition.enable" = "true",
  "dynamic_partition.time_unit" = "MONTH",
  "dynamic_partition.start" = "-12",
  "dynamic_partition.end" = "1",
  "dynamic_partition.prefix" = "p",
  "dynamic_partition.create_history_partition"="true",
  "replication_num" = "3"
);

When I use Kafka to synchronize data to doris，The AUTO PARTITION is changed：
AUTO PARTITION BY RANGE date_trunc(`B`.null, 'MONTH')
(PARTITION p202303 VALUES [('2023-03-01'), ('2023-04-01')),
PARTITION p202304 VALUES [('2023-04-01'), ('2023-05-01')),
PARTITION p202305 VALUES [('2023-05-01'), ('2023-06-01')),
PARTITION p202306 VALUES [('2023-06-01'), ('2023-07-01')),
PARTITION p202307 VALUES [('2023-07-01'), ('2023-08-01')),
PARTITION p202308 VALUES [('2023-08-01'), ('2023-09-01')),
PARTITION p202309 VALUES [('2023-09-01'), ('2023-10-01')),
PARTITION p202310 VALUES [('2023-10-01'), ('2023-11-01')),
PARTITION p202311 VALUES [('2023-11-01'), ('2023-12-01')),
PARTITION p202312 VALUES [('2023-12-01'), ('2024-01-01')),
PARTITION p202401 VALUES [('2024-01-01'), ('2024-02-01')),
PARTITION p202402 VALUES [('2024-02-01'), ('2024-03-01')),
PARTITION p202403 VALUES [('2024-03-01'), ('2024-04-01')),
PARTITION p202404 VALUES [('2024-04-01'), ('2024-05-01')))
I WOULD LIKE TO ASK WHY AUTO PARTITION IS CHANGING？

### What You Expected?

how to change AUTO PARTITION with ‘AUTO PARTITION BY RANGE date_trunc(reportDateTime,'MONTH')’ and why this wrong happend？

### How to Reproduce?

Kafka to doris send about 40000000 message

### Anything Else?

none

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by zclllyybb** [comment]
hi, can you add my wechat torch-wood_ for more information?

**Comment by someorz** [comment]
我也遇到了，分区字段会变，AUTO PARTITION 我使用的是list方式，写入一些数据后分区字段变了，很奇怪的是，'dr'.'dt'，dr其实是接口表名的别名。

**Comment by mark-triker** [comment]
do your solve the problem? i have met the same problem

**Comment by zclllyybb** [comment]
> do your solve the problem? i have met the same problem

yes. this bug was fixed since doris-2.1.4. for the version before it, disable falling back to old planner will avoid it.
